src/plugins/dashboard/public/dashboard_app/dashboard_app.test.tsx

```diff
@@ -0,0 +1,105 @@
+import { DashboardApi } from '..';
+
+/* These tests circumvent the need to test the router and legacy code
+/* the dashboard app will be passed the expanded panel id from the DashboardRouter through mountApp()
+/* @link https://github.com/elastic/kibana/pull/190086/
+*/
+
+describe('Dashboard App', () => {
+  const mockDashboard = buildMockDashboard();
+  const expandedPanelIdSpy = jest.spyOn(mockDashboard.expandedPanelId, 'next');
+  let mockHistory: MemoryHistory;
+  let historySpy: jest.SpyInstance;
+  beforeAll(() => {
+    mockHistory = createMemoryHistory();
+    historySpy = jest.spyOn(mockHistory, 'replace');
+    /**
+     * Mock the LazyDashboardRenderer component to avoid rendering the actual dashboard
+     * and hitting errors that aren't relevant
+     */
+    jest
+      .spyOn(dashboardRendererStuff, 'LazyDashboardRenderer')
+      // we need overwrite the onApiAvailable prop to get the dashboard Api in the dashboard app
+      .mockImplementation(({ onApiAvailable }: DashboardRendererProps) => {
+        useEffect(() => {
+          onApiAvailable?.(mockDashboard as DashboardApi);
+        }, [onApiAvailable]);
+
+        return <div>Test renderer</div>;
+      });
+  });
+
+  beforeEach(() => {
+    // reset the mock history calls per test
+    historySpy.mockClear();
+    // reset the expanded panel id between tests
+    expandedPanelIdSpy.mockClear();
+  });
+
+  it('test the default behavior without an expandedPanel id passed as a prop to the DashboardApp', async () => {
+    // default conditions with no expanded panel id and no urls being passed with expanded panel ID in the url
+    render(<DashboardApp redirectTo={jest.fn()} history={mockHistory} />);
+
+    await waitFor(() => {
+      // this value should be undefined by default
+      expect(mockDashboard.expandedPanelId.getValue()).toBe(undefined);
+      // history should not be called
+      expect(historySpy).toHaveBeenCalledTimes(0);
+      expect(mockHistory.location.pathname).toBe('/');
+    });
+
+    // simulate expanding a panel
+    mockDashboard.expandPanel('123');
+
+    await waitFor(() => {
+      expect(mockDashboard.expandedPanelId.getValue()).toBe('123');
+      expect(historySpy).toHaveBeenCalledTimes(1);
+      expect(mockHistory.location.pathname).toBe('/create/123');
+    });
+  });
+
+  it('test that the expanded panel behavior subject and history is called when passed as a prop to the DashboardApp', async () => {
+    // this tests a user getting a url from another user that has an expanded panel on the dashboard
+    // test someone giving a url with an expanded panel id, as if it goes through the router
+    mockHistory.push('/create/456');
+
+    render(
+      <DashboardApp
+        savedDashboardId=""
+        redirectTo={jest.fn()}
+        history={mockHistory}
+        expandedPanelId="456"
+      />
+    );
+
+    await waitFor(() => {
+      expect(expandedPanelIdSpy).toHaveBeenCalledTimes(1);
```

Comment:
I think spying on `next` is actually a bit confusing - because we're ultimately trying to test if the **subscription** fires, not if `next` is called. I think it's better to check the value of `expandedPanelId`?